library/kani_macros/src/sysroot/contracts/mod.rs

```diff
@@ -400,36 +488,35 @@ pub fn proof_for_contract(attr: TokenStream, item: TokenStream) -> TokenStream {
     .into()
 }
 
-/// Classifies the state a function is in in the contract handling pipeline.
+/// Classifies the state a function is in the contract handling pipeline.
 #[derive(Clone, Copy, PartialEq, Eq)]
 enum ContractFunctionState {
-    /// This is the original code, re-emitted from a contract attribute.
-    Original,
+    /// This is the function already expanded with the closures.
+    Expanded,
     /// This is the first time a contract attribute is evaluated on this
     /// function.
     Untouched,
-    /// This is a check function that was generated from a previous evaluation
-    /// of a contract attribute.
-    Check,
-    /// This is a replace function that was generated from a previous evaluation
-    /// of a contract attribute.
-    Replace,
-    ModifiesWrapper,
 }
 
 /// The information needed to generate the bodies of check and replacement
 /// functions that integrate the conditions from this contract attribute.
 struct ContractConditionsHandler<'a> {
-    function_state: ContractFunctionState,
     /// Information specific to the type of contract attribute we're expanding.
     condition_type: ContractConditionsData,
     /// Body of the function this attribute was found on.
-    annotated_fn: &'a mut ItemFn,
+    annotated_fn: &'a ItemFn,
     /// An unparsed, unmodified copy of `attr`, used in the error messages.
     attr_copy: TokenStream2,
     /// The stream to which we should write the generated code.
     output: TokenStream2,
-    hash: Option<u64>,
+    /// The name of the check closure.
+    check_name: String,
+    /// The name of the replace closure.
+    replace_name: String,
+    /// The name of the recursion closure.
+    recursion_name: String,
+    /// The name of the modifies closure.
+    modify_name: String,
```

Comment:
sometimes we refer as inner check sometimes as modify name, should we be more consistent here?